src/main/java/games/strategy/util/Version.java

```diff
@@ -64,6 +66,20 @@ public Version(final String version) {
     }
   }
 
+  /**
+   * Returns the exact and full version number.
+   * For example, if we specify:
+   * <code>
+   * new Version(1.2.3.4.5).getMicro == 4; // true
+   * new Version(1.2.3.4.5).toString().equals("1.2.3.4"); // true
+   * new Version(1.2.3.4.5).getExactVersion.equals("1.2.3.4.5"); // true
+   * </code>
+   */
+  public String getExactVersion() {
+    // in case of deserialization, exactVersion may be null, in which case toString() it.
+    return exactVersion != null ? exactVersion : toString();
```

Comment:
Would it be cleaner to separate the deserialization concern and handle this case in `readObject()`?  For example:

```java
private void readObject(final ObjectInputStream in) throws IOException, ClassNotFoundException {
  in.defaultReadObject();

  if (exactVersion == null) {
    exactVersion = toString();
  }
}
```

I'm not 100% sure about calling `toString()` from within `readObject()`, but it should be ok as the required fields should be initialized by the time `in.defaultReadObject()` returns.